portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,19 +30,25 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-
-
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/reviews")
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> reviews = new ArrayList<String>();
-
   @Override
   public void init() {
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query reviewquery = new Query("Task").addSort("review", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery allreviews = datastore.prepare(reviewquery);
+    ArrayList<String> reviews = new ArrayList<String>();
+
+    for (Entity review : allreviews.asIterable()) {
+      String reviewoutput = (String) review.getProperty("review");
```

Comment:
What happens if this cast fails? can it fail?

Reply:
if it fails it will just throw exception. I use cast instead of .tostring() in order to validate my assumptions that input is string, otherwise I might be converting some garbo input into string which is bad

Reply:
Ah makes sense! Throwing an exception seems fine then.